### ISSUE TITLE
[selectedOptions] is not shown at the autocomplete list for mat-selectionlist

### ISSUE THREAD

**Issue by Omcsesz** [angular/vscode-ng-language-service#1836]
# 🐞 bug report

### Is this a regression?
No, it didn't work previously.

### Description

 A clear and concise description of the problem...

Hello, Team! 

I'm using Angular Material version 14.2.7. According to the official documentation (https://v14.material.angular.io/components/list/api#MatSelectionList), the mat-selection-list has a property called selectedOptions, but it is not shown in the autocomplete suggestions (see attached gif). Also, the property selected is not not shown for mat-list-option. Can you please fix this?
[demoProject2.zip](https://github.com/angular/vscode-ng-language-service/files/10202407/demoProject2.zip)
[AutocompleteForAngularNotWorkingCorrectly](https://user-images.githubusercontent.com/1839739/206909444-3eb88867-23db-47b1-9490-a385ced6db56.gif)

## Bug Type
What does this bug affect

- [X] Angular Language Service VSCode extension
- [X] Angular Language Service server

## Reproduction

1. Open the demo project I provided in the attached zip file
2. Run npm install
3. Open src\app\shared\inputs\checkbox-group\checkbox-group.component.html
4. Try to add property selectedOptions to mat-selection-list, or selected to mat-list-option

**Expected behavior**

The property `selectedOptions` should be in the popup list for `mat-selection-list`, and the property `selected` should be in the popup list for `mat-list-option`.

### **Logs**

Log file gives us deep insight into the behavior and performance of the extension. If the issue is a performance
problem or an error occured, please provide the output of the log file below.

Set Angular Log level to verbose.

<pre><code>

</code></pre>

### **Screenshots**

![Example Screenshot](./output.png)

## 🌍  Your Environment
**Angular Version:**
<pre><code>
Angular CLI: 15.0.2
Node: 18.12.1
Package Manager: npm 9.1.3
OS: win32 x64

Angular: 15.0.2
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.1500.2
@angular-devkit/build-angular      15.0.2
@angular-devkit/core               15.0.2
@angular-devkit/schematics         15.0.2
@angular/cdk                       14.2.7
@angular/material                  14.2.7
@angular/material-moment-adapter   14.2.7
@schematics/angular                15.0.2
rxjs                               6.6.7
typescript                         4.8.4
</code></pre>

Extension Version:
<pre><code>Angular Language Service v15.0.2</code></pre>

VSCode Version:
<pre><code>1.74.0 (user setup)</code></pre>

Operating System:
<pre><code>Windows 10 build19045</code></pre>

Extension options:
<pre><code>"angular.enable-strict-mode-prompt": false</code></pre>

**Comment by atscott** [comment]
The listed properties are neither inputs nor outputs of their components. Closing as working as intended.

**Comment by Omcsesz** [comment]
> The listed properties are neither inputs nor outputs of their components. Closing as working as intended.

Should they be inputs or outputs in order to be on the autocomplete list?

**Comment by atscott** [comment]
> Should they be inputs or outputs in order to be on the autocomplete list?

Yes

**Comment by atscott** [comment]
> Good, because [selected] is actually an input:

Indeed, looks like it is. Your project is not using `strictTemplates` so the issue is actually that `mat-list-option` is part of a template that's not type checked because of your project's template type checking configuration. If you cannot update to `strictTemplates: true`, I would recommend using the extension option to override it:

![image](https://user-images.githubusercontent.com/479713/207124279-7dabdff2-d179-4af8-8ffe-0cfbcf3c142a.png)
